lib/InputTest.ts

```diff
@@ -57,6 +57,26 @@ export declare interface InputTest {
     event: InputTest.Events.Volume,
     value: number,
   ): boolean;
+  /**
+   * This event is emitted when the test encounters a non-fatal warning during
+   * its run-time.
+   * @param event [[InputTest.Events.Warning]]
+   * @param warning The warning that the test encountered.
+   */
+  emit(
+    event: InputTest.Events.Warning,
+    warning: WarningName,
+  ): boolean;
+  /**
+   * This event is emitted when the test clears a previously encountered
+   * non-fatal warning during its run-time.
+   * @param event [[InputTest.Events.WarningCleared]]
+   * @param warning The warning that the test encountered that should be cleared.
+   */
+  emit(
```

Comment:
Should we tag these two new emits as `@private`? I think we are only exposing the `.on()` right?